backend/utils/filesystem.py

```diff
@@ -27,3 +28,35 @@ def iter_directories(path: str, recursive: bool = False) -> Iterator[tuple[Path,
             yield Path(root), directory
         if not recursive:
             break
+
+
+INVALID_CHARS = r'[\\/:*?"<>|]'
```

Comment:
We aren't using this constant. Should we replace this by two constants, one for characters that get replaced by hyphen, and another one for characters that get removed?

Also, we can use `re.compile`, same as other regex constants.